core/src/main/scala/com/softwaremill/session/CsrfDirectives.scala

```diff
@@ -36,6 +40,15 @@ trait CsrfDirectives {
     }
   }
 
+  /**
+   * Java alternative
+   */
+  def randomTokenCsrfProtection[T](checkMode: CsrfCheckMode[T], inner: Supplier[Route]): Route = RouteAdapter {
```

Comment:
is this needed here, can't the code live in `javadsl`? If it's here, users of the scalads will also see the java variants which might be confusing